### PR TITLE
[BD-21] Get rid of the `documented_elsewhere` annotation

### PR DESCRIPTION
**Pull request by regisb**
**Description:** This is nowhere used. If we need to re-introduce it at some point, we'll
discuss it then.

**JIRA:** BD-21 https://openedx.atlassian.net/wiki/spaces/COMM/pages/1596358943

**Reviewers:**
- [x] @robrap 

**Merge checklist:**
- [x] All reviewers approved
- [x] CI build is green
- [x] Version bumped
- [x] Changelog record added
- [ ] ~~Documentation updated (not only docstrings)~~
- [x] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant
solutions, hacks, quick-and-dirty implementations, concerns about
migrations, etc.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @regisb! I've created [BLENDED-604](https://openedx.atlassian.net/browse/BLENDED-604) to keep track of it in Jira.   More details are on the [BD-21](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1596358943/BD-21+Toggles+Settings+Documentation) project page.


When this pull request is ready, tag your edX technical lead.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -11,6 +11,12 @@ Change Log
 
 .. There should always be an "Unreleased" section for changes pending release.
 
+[0.9.0] - 2020-09-25
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Get rid of the ``documented_elsewhere`` annotation, which is used nowhere
```
Comment (robrap) [inline CHANGELOG.rst:17]: Not sure if we need to split deprecation and removal and use a major version change when we are ready? I always get burned when I try to cheat the system. Here are some references: https://github.com/search?q=documented_elsewhere&type=code

Comment (regisb) [reply CHANGELOG.rst:17]: AFAIK this is only used in edx-lint: https://github.com/search?q=org%3Aedx+documented_elsewhere&type=code

Here is the PR to get rid of it: https://github.com/edx/edx-lint/pull/127

Comment (robrap) [reply CHANGELOG.rst:17]: Sounds good.

**Review by robrap — APPROVED** [review]

**Pull request merged**